cli/packages/util/keyringwrapper.go

```diff
@@ -20,16 +23,51 @@ func SetValueInKeyring(key, value string) error {
 		PrintErrorAndExit(1, err, "Unable to get current vault. Tip: run [infisical rest] then try again")
 	}
 
-	return keyring.Set(currentVaultBackend, MAIN_KEYRING_SERVICE, key, value)
+	err = keyring.Set(currentVaultBackend, MAIN_KEYRING_SERVICE, key, value)
+
+	if err != nil {
+		configFile, _ := GetConfigFile()
+		PrintWarning("Fallback file keyring is being used\n\nYou can persist your file passphrase by running the following command:\ninfisical vault set file --passphrase <your-passphrase>\n")
+
+		if configFile.VaultBackendPassphrase == "" {
+			passphrasePrompt := promptui.Prompt{
+				Label: "Enter the passphrase to use for keyring encryption: ",
+			}
+			passphrase, err := passphrasePrompt.Run()
```

Comment:
I'm getting some kind of massive lag when typing in the terminal

Reply:
I haven't gotten this before no -- how big is your terminal size? I know our CLI generally gets very buggy when the terminal is small

Reply:
figured it out